lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -172,6 +189,16 @@ Type TypeChecker::getOrCreateOpaqueResultType(TypeResolution resolution,
     return existingDecl->getDeclaredInterfaceType();
   }
   
+  // Check the availability of the opaque type runtime support.
+  auto runningOS = overApproximateAvailabilityAtLocation(repr->getLoc(),
```

Comment:
This should be done in DeclChecker::visitFuncDecl() instead of here, so that the checking only runs in primary files.

Reply:
In other cases, we check the availability of types during type repr resolution. Since this function overrides type repr resolution for OpaqueReturnTypeReprs, it should happen in the same cases where we would've checked the availability of another kind of type reference. Would it be sufficient to test this by building a two-file project and checking that no errors are emitted in the primary file when the other file has unavailable opaque return types?

Reply:
It looks like we re-diagnose availability in non-primary files for other types too.

Reply:
Ok, you're right. We've discussed moving this all to a visitor that runs on primary files only but that's a big change. Never mind then!